graylog2-server/src/main/java/org/graylog2/cluster/lock/MongoLockService.java

```diff
@@ -98,23 +99,30 @@ private void ensureTTLIndex(MongoCollection<Document> collection, Bson updatedAt
         collection.createIndex(updatedAtKey, indexOptions);
     }
 
-    /**
-     * Request a lock. If a lock already exists, the lock expiry time will be extended.
-     *
-     * @param resource Unique identifier for the resource that should be guarded by this lock.
-     * @return A {@link Lock} object, if a lock was obtained. An empty {@link Optional}, if no lock could be acquired.
-     */
     @Override
-    public Optional<Lock> lock(@Nonnull String resource) {
+    public Optional<Lock> lock(@Nonnull String resource, @Nullable String lockContext) {
+        if (lockContext == null) {
+            return doLock(resource, nodeId.toString());
+        }
+        return doLock(resource, nodeId.toString() + "-" + lockContext);
+    }
+
+    @Override
+    public Optional<Lock> extendLock(@Nonnull Lock lock) {
```

Comment:
We should probably add a null check here to avoid an NPE if `lock` is null.

Reply:
sure. I wonder how much benefit we get from `@Nonnull` though. IntelliJ claims it can never be null :-D

Reply:
I think we only benefit from it in the IDE when we try to pass `null` to it at some point. Some static analysis tools could also use it.